libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -405,11 +411,16 @@ public void checkWriteProperty(Class<?> callerClass, String property) {
                 requestingClass.getModule().getName(),
                 requestingClass,
                 property
-            )
+            ),
+            callerClass
         );
     }
 
-    private static void notEntitled(String message) {
+    private static void notEntitled(String message, Class<?> callerClass) {
+        // don't log self tests in EntitlementBootstrap
+        if (EntitlementBootstrap.class.equals(callerClass) == false) {
+            logger.warn(message);
```

Comment:
If you construct the not entitled exception first then the exception can be logged, along with the stack trace.

Reply:
I thought about doing it, but decided against it as we'd be logging the exception twice most of the time. I often find that more confusing / disturbing than being helpful in rare cases. On the other hand, obviously, those warnings are a lot more noticeable if there's a stacktrace attached.
I don't have a strong preference, happy to follow the crowed here.

Reply:
I don't feel strongly in this case. The message is likely to have enough info to determine the remedy already.

Reply:
Same, not a strong preference. OK to leave it as-is -- like you said, this is mainly for when we have code that catches Throwable or similar.

Reply:
👍 added